src/docker-compose-cli.js

```diff
@@ -31,7 +34,16 @@ const composeCommand = (filePaths, ...params) => {
   });
 };
 
-const pull = (fileName) => composeCommand(fileName, 'pull');
+const pull = async (fileName) => {
+  try {
+    await composeCommand(fileName, 'pull');
+  } catch (err) {
+    if (isRateExceededError(err)) {
```

Comment:
Should we be concerned about an infinite loop here?  What if we just keep getting `429`s for some reason?  At least might be good to add some kind of logging that we are retrying so someone checking the logs can see where it is stuck....

Reply:
I added a 100 retry limit.